Sources/Hyperwallet.swift

```diff
@@ -56,6 +56,20 @@ public final class Hyperwallet: NSObject {
         instance = Hyperwallet(provider)
     }
 
+    /// Creates a new instance of the Hyperwallet Core SDK interface object. If a previously created instance exists,
+    /// it will be replaced. Next it retrieves the authentication token from the provider and decodes the token in
+    /// case of successful response else returns error
+    ///
+    /// - Parameters:
+    ///   - provider: a provider of Hyperwallet authentication tokens.
+    ///   - completion: the callback handler of responses from the Hyperwallet platform
+    public class func setup(_ provider: HyperwalletAuthenticationTokenProvider,
```

Comment:
this class 'Hyperwallet' is final. We should use the static function instead of the class function.

Reply:
This function can be static, but not because class is final. Final classes can have non-static methods.

Reply:
@ymoskovko-epam could you provide some reasoning for this? The class is final meaning it cannot be subclassed. https://stackoverflow.com/questions/29636633/static-vs-class-functions-variables-in-swift-classes mentions that the difference is a matter of preference - but didnt see any reason why the function needs to be static

Reply:
static func == final class func

Our class is final and we use both static and class declarations for functions. Let's use one style

Reply:
I would do this in a different PR - which needs to update the documentation and make sure that the integration with ui-sdk is working properly. I would say lets do this along with the ui sdk insight changes